adapters/contxtful/contxtful.go

```diff
@@ -0,0 +1,562 @@
+		request:          request,
+		requestData:      requestData,
+		bidderCustomerId: bidderCustomerId,
+		bidderVersion:    bidderVersion,
+		bidderResponse:   adapters.NewBidderResponse(),
+		domain:           domain,
+		adRequestID:      request.ID,
+	}
+
+	if a.processResponse(response.Body, ctx) {
+		return ctx.bidderResponse, ctx.errors
+	}
+
+	return nil, []error{fmt.Errorf("failed to parse response as Contxtful relay or Prebid.js format")}
+}
+
+// Simplified bid creation using prebid bid struct
+func (a *adapter) createBid(
+	prebidBid ContxtfulExchangeBid,
+	currency string,
+	ctx *BidProcessingContext,
+	customerId string,
+) {
+	// Determine media type from impression
+	var bidType openrtb_ext.BidType = openrtb_ext.BidTypeBanner
+	for _, imp := range ctx.request.Imp {
+		if imp.ID == prebidBid.RequestID {
+			if imp.Video != nil {
+				bidType = openrtb_ext.BidTypeVideo
+			} else if imp.Native != nil {
+				bidType = openrtb_ext.BidTypeNative
+			}
+			break
+		}
+	}
+
+	bid := &openrtb2.Bid{
+		ID:    fmt.Sprintf("%s-%s", BidderName, prebidBid.RequestID),
+		ImpID: prebidBid.RequestID,
+		Price: prebidBid.CPM,
+		AdM:   prebidBid.AdM,
+		W:     int64(prebidBid.Width),
+		H:     int64(prebidBid.Height),
+		CrID:  prebidBid.CreativeID,
+		NURL:  prebidBid.NURL,
+		BURL:  prebidBid.BURL,
+		LURL:  prebidBid.LURL,
+		Ext:   prebidBid.Ext,
+	}
+
+	typedBid := &adapters.TypedBid{
+		Bid:     bid,
+		BidType: bidType,
+	}
+
+	ctx.bidderResponse.Bids = append(ctx.bidderResponse.Bids, typedBid)
+}
+
+// Simple unified bidder config extraction - get ORTB2 data and extract params
+func extractBidderConfig(request *openrtb2.BidRequest) (string, string) {
+	if request == nil || request.Ext == nil {
+		return "", ""
+	}
+
+	var requestExt openrtb_ext.ExtRequest
+
+	if err := jsonutil.Unmarshal(request.Ext, &requestExt); err != nil {
+		return "", ""
+	}
+
+	// Find contxtful bidder config and extract params from ORTB2 data
+	for _, config := range requestExt.Prebid.BidderConfigs {
+		for _, bidder := range config.Bidders {
+			if bidder == BidderName && config.Config.ORTB2 != nil {
+				return extractContxtfulParams(config.Config.ORTB2)
+			}
+		}
+	}
+	return "", ""
+}
+
+// Extract contxtful params from any ORTB2 data (unified for both bidder config and other sources)
+func extractContxtfulParams(ortb2Data *openrtb_ext.ORTB2) (string, string) {
+	var ortb2UserData []struct {
+		Name string `json:"name"`
+		Ext  struct {
+			Params struct {
+				CI string `json:"ci"` // Customer ID
+				EV string `json:"ev"` // Version
+			} `json:"params"`
+		} `json:"ext"`
+	}
+
+	if err := jsonutil.Unmarshal(ortb2Data.User, &ortb2UserData); err != nil {
+		return "", ""
+	}
+
+	// Find contxtful params
+	for _, data := range ortb2UserData {
+		if data.Name == BidderName && data.Ext.Params.CI != "" {
+			version := data.Ext.Params.EV
+			if version == "" {
+				version = DefaultVersion
+			}
+			return data.Ext.Params.CI, version
+		}
+	}
```

Comment:
this block needs a test

Reply:
I also removed `extractRequestConfig` since customerId is always available in the bid params.
see commit c0e2052253b069f4dd5af79495944367c9823914

The existing test [TestInvalidParams](https://github.com/prebid/prebid-server/blob/c0e2052253b069f4dd5af79495944367c9823914/adapters/contxtful/params_test.go#L23) tests for cases with missing customerId or missing placementId.